File: Demo/Sources/ViewControllers/ActivityView/ActivityViewController.m
//
//  Globus
//

#import "ActivityViewController.h"

@interface ActivityViewController ()

@property(nonatomic, weak) IBOutlet GLBListField* listField;
@property(nonatomic, weak) IBOutlet GLBButton* showButton;

@property(nonatomic, strong) GLBTimer* timer;

@end

@implementation ActivityViewController

#pragma mark - UIViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    _timer = [GLBTimer timerWithInterval:5.0f];
    _timer.actionFinished = [GLBAction actionWithTarget:self action:@selector(timerFinished)];
    
    _listField.items = @[
        [GLBListFieldItem itemWithTitle:GLBArcSpinnerView.glb_className value:GLBArcSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBArcAltSpinnerView.glb_className value:GLBArcAltSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBBounceSpinnerView.glb_className value:GLBBounceSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBChasingDotsSpinnerView.glb_className value:GLBChasingDotsSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBCircleSpinnerView.glb_className value:GLBCircleSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBCircleFlipSpinnerView.glb_className value:GLBCircleFlipSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBFadingCircleSpinnerView.glb_className value:GLBFadingCircleSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBFadingCircleAltSpinnerView.glb_className value:GLBFadingCircleAltSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBNineCubeGridSpinnerView.glb_className value:GLBNineCubeGridSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBPlaneSpinnerView.glb_className value:GLBPlaneSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBPulseSpinnerView.glb_className value:GLBPulseSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBThreeBounceSpinnerView.glb_className value:GLBThreeBounceSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBWanderingCubesSpinnerView.glb_className value:GLBWanderingCubesSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBWaveSpinnerView.glb_className value:GLBWaveSpinnerView.class],
        [GLBListFieldItem itemWithTitle:GLBWordPressSpinnerView.glb_className value:GLBWordPressSpinnerView.class],
    ];
    
    self.activityView = [GLBActivityView new];
    self.activityView.spinnerView = GLBArcSpinnerView.new;
    self.activityView.textLabel.text = @"This is activity message";
}

#pragma mark - Actions

- (IBAction)pressedMenu:(id)sender {
    [self.glb_slideViewController showLeftViewControllerAnimated:YES complete:nil];
}

- (IBAction)pressedShow:(id)sender {
    [_listField resignFirstResponder];
    [self.activityView show];
    [_timer start];
}

- (IBAction)changedSelectedItem:(id)sender {
    GLBListFieldItem* selected = _listField.selectedItem;
    if(selected != nil) {
        Class spinnerViewClass = selected.value;
        if(selected != nil) {
            self.activityView.spinnerView = spinnerViewClass.new;
        } else {
            self.activityView.spinnerView = nil;
        }
    } else {
        self.activityView.spinnerView = nil;
    }
}

#pragma mark - Timer

- (void)timerFinished {
    [self.activityView hideComplete:^{
        [_listField becomeFirstResponder];
    }];
}

@end

File: Demo/Sources/ViewControllers/ActivityView/ActivityViewController.swift
//
//  Globus
//

import Globus

class ActivityViewController: GLBViewController {
    
    // MARK - Outlet property
    
    @IBOutlet fileprivate weak var listField: GLBListField!
    @IBOutlet fileprivate weak var showButton: GLBButton!
    
    private var timer: GLBTimer?
    
    // MARK - UIViewController
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.timer = GLBTimer(interval: 5.0);
        self.timer!.actionFinished = GLBAction(target: self, action: #selector(timerFinished));

        self.listField.items = [
            GLBListFieldItem(title: GLBArcSpinnerView.glb_className(), value:GLBArcSpinnerView.self),
            GLBListFieldItem(title: GLBArcAltSpinnerView.glb_className(), value:GLBArcAltSpinnerView.self),
            GLBListFieldItem(title: GLBBounceSpinnerView.glb_className(), value:GLBBounceSpinnerView.self),
            GLBListFieldItem(title: GLBChasingDotsSpinnerView.glb_className(), value:GLBChasingDotsSpinnerView.self),
            GLBListFieldItem(title: GLBCircleSpinnerView.glb_className(), value:GLBCircleSpinnerView.self),
            GLBListFieldItem(title: GLBCircleFlipSpinnerView.glb_className(), value:GLBCircleFlipSpinnerView.self),
            GLBListFieldItem(title: GLBFadingCircleSpinnerView.glb_className(), value:GLBFadingCircleSpinnerView.self),
            GLBListFieldItem(title: GLBFadingCircleAltSpinnerView.glb_className(), value:GLBFadingCircleAltSpinnerView.self),
            GLBListFieldItem(title: GLBNineCubeGridSpinnerView.glb_className(), value:GLBNineCubeGridSpinnerView.self),
            GLBListFieldItem(title: GLBPlaneSpinnerView.glb_className(), value:GLBPlaneSpinnerView.self),
            GLBListFieldItem(title: GLBPulseSpinnerView.glb_className(), value:GLBPulseSpinnerView.self),
            GLBListFieldItem(title: GLBThreeBounceSpinnerView.glb_className(), value:GLBThreeBounceSpinnerView.self),
            GLBListFieldItem(title: GLBWanderingCubesSpinnerView.glb_className(), value:GLBWanderingCubesSpinnerView.self),
            GLBListFieldItem(title: GLBWaveSpinnerView.glb_className(), value:GLBWaveSpinnerView.self),
            GLBListFieldItem(title: GLBWordPressSpinnerView.glb_className(), value:GLBWordPressSpinnerView.self),
        ]
        
        self.activityView = GLBActivityView();
        self.activityView!.spinnerView = GLBArcSpinnerView()
        self.activityView!.textLabel!.text = "This is activity message"
    }
    
    // MARK - Action
    
    @IBAction internal func pressedMenu(_ sender: Any) {
        self.glb_slideViewController?.showLeftViewController(animated: true, complete: nil)
    }
    
    @IBAction internal func pressedShow(_ sender: Any) {
        if let activityView = self.activityView {
            self.listField.resignFirstResponder()
            activityView.show()
            self.timer!.start()
        }
    }
    
    @IBAction internal func changedSelectedItem(_ sender: Any) {
        if let selected = self.listField.selectedItem {
            if let spinnerViewClass = selected.value as? GLBSpinnerView.Type {
                self.activityView!.spinnerView = spinnerViewClass.init()
            } else {
                self.activityView!.spinnerView = nil
            }
        } else {
            self.activityView!.spinnerView = nil
        }
    }
    
    // MARK: - Timer
    
    @objc private func timerFinished() {
        self.activityView!.hideComplete {
            self.listField.becomeFirstResponder()
        }
    }
    
    // MARK: - GLBNibExtension
    
    public override static func nibName() -> String {
        return "ActivityViewController-Swift"
    }
    
}
